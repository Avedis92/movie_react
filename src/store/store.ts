import { configureStore } from '@reduxjs/toolkit';
import genreReducer from './slices/genre';

export const store = configureStore({
  reducer: {
    genre: genreReducer,
  },
});

export type RootState=ReturnType<typeof store.getState>;
export type MovieDispatch=typeof store.dispatch;
