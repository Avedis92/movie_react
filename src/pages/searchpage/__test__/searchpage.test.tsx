import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { store } from '../../../store/store';
import '@testing-library/jest-dom/extend-expect';
import SearchPage from '../searchpage';

jest.mock('react-router-dom', () => ({
  useLocation: jest.fn().mockImplementation(() => ({
    pathname: '/search?searchedMovie=pirates',
  })),
  useHistory: jest.fn().mockImplementation(() => ({
    pathname: `/moviedetailpage?movieId=${550}` || '/',
  })),
}));

describe('Make tests on SearchPage component', () => {
  test('Make a snapshot test', () => {
    const {asFragment} = render(
      <Provider store={store}>
        <SearchPage />
      </Provider>,
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
