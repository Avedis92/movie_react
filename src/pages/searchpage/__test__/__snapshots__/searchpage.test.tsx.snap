// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Make tests on SearchPage component Make a snapshot test 1`] = `
<div
  className="Container"
>
  <div
    className="searchHeader"
  >
    <h1
      onClick={[Function]}
    >
      Movies
    </h1>
    <input
      list="movieTitle"
      name="movieSearch"
      onKeyPress={[Function]}
      placeholder="Search"
      type="text"
    />
  </div>
  <div
    className="searchElement"
    style={
      Object {
        "backgroundImage": "url(https://cdn3.vectorstock.com/i/1000x1000/67/12/404-error-page-background-vector-20596712.jpg",
      }
    }
  >
    <div
      className="searchFooter"
    />
  </div>
</div>
`;
